elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/ElasticsearchHSQueryImpl.java

```diff
@@ -694,6 +696,16 @@ private SearchResult search(boolean enableScrolling) {
 				}
 			}
 			Search search = builder.build();
+
+			if ( QUERY_LOG.isDebugEnabled() ) {
+				try ( ServiceReference<GsonService> gsonService =
+						extendedIntegrator.getServiceManager().requestReference( GsonService.class ) ) {
```

Comment:
> Think in terms of limits: what would happen on a single threaded app? For every single query, we'd be starting the service, then stopping it again.

Ok, that not what I had in mind. It's true in the general case, though even in a single threaded app, `GsonService` is likely to be referenced elsewhere, so it likely wouldn't be started for each query. It is for instance referenced by `org.hibernate.search.elasticsearch.client.impl.JestClient`, which is referenced (indirectly) by `ElasticsearchIndexManager`, whose lifespan is large enough to make this start/stop dance quite rare.

Anyway... We're both willing to avoid this potential performance issue here. I'll simply repeat what I said yesterday: in that particular case, there's no single, non-service object created when bootstrapping that could hold the reference to this service and pass it along to queries at runtime. I would _like_ to do so, but it's simply not possible in the current state of the code. Or at least I'm unable to see how it's possible. Which is why I'm trying to think of alternative solutions...

I created a ticket to address potential performance issues in the `ServiceManager` for those few cases where we are forced to repeatedly acquire reference to services. We should avoid it, I agree, but there may be cases where we can't do otherwise. See https://hibernate.atlassian.net/browse/HSEARCH-2504